auth/auth.go

```diff
@@ -0,0 +1,27 @@
+package auth
+
+type key string
+
+const (
+	// The current auth configuration that has been set by any auth middleware and
+	// that will prevent from being set more than once.
+	CURRENT_AUTH_CONFIG key = "currentAuthConfig"
```

Comment:
Is this a CURRENT_AUTH_CONFIG or CURRENT_API_KEY_AUTH_CONFIG?
Is this actually going to be set by _any_ auth middleware? sigv4 middleware?

Reply:
This auth config should be used any auth middleware to prevent us from setting the auth more than once (e.g. you would send Bearer token in the Authorization header and the API key in the query parameter in the same request). I think sigv4 middleware would not require us setting this config, because it is a special case. For it, I believe we would only allow one auth middleware (for a particular operation) in the codegen itself. I don't believe we can design an operation in the Smithy model that can allow sigv4 with any other auth mechanism, right?